server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -407,6 +422,184 @@ public TimeValue timeout() {
         }, "create_snapshot [" + snapshotName + ']', listener::onFailure);
     }
 
+    /**
+     * Initializes the snapshotting process for clients when Snapshot v2 is enabled. This method is responsible for taking
+     * a shallow snapshot and pinning the snapshot timestamp.The entire process is executed on the cluster manager node.
+     *
+     * Unlike traditional snapshot operations, this method performs a synchronous snapshot execution and doesn't
+     * upload any shard metadata to the snapshot repository.
+     * The pinned timestamp is later reconciled with remote store segment and translog metadata files during the restore
+     * operation.
+     *
+     * @param request  snapshot request
+     * @param listener snapshot creation listener
+     */
+    public void createSnapshotV2(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
```

Comment:
How are we ensuring that this method is synchronised with the cleanup?

Reply:
Didn't get this. Do you mean cleanup of remote store metadata files?